src/generic/FreeAssAlgebraGroebner.jl

```diff
@@ -0,0 +1,641 @@
+###############################################################################
+#
+#   FreeAssAlgebraGroebner.jl : free associative algebra R<x1,...,xn> Groebner basis
+#
+###############################################################################
+
+export groebner_basis
+
+export interreduce!
+
+export normal_form
+
+export normal_form_weak
+
+import DataStructures: PriorityQueue, enqueue!, dequeue!
+
+const groebner_debug_level = 0
+const Monomial = Vector{Int}
```

Comment:
The reasoning was that the Aho-Corasick algorithm could also be used for different purposes than just for monomials in algebras, and since it is traditionally used for substring searching, I used Word. However I would not mind changing it to Monomial in the AhoCorasick.jl file, if you would prefer that.

Reply:
I don't mind having two types, as long as it is clear for the reader *why* there are two, so maybe add a comment?

(In the long run, it would probably be wise to actually allow a parametrized type -- unless you have have many algebras with more than 2^31 generators, it probably would be useful to allow for words over Int32 (or rather `UInt32`, or even `UInt16` and `UInt8`). That reduces memory usage, thus reduces cache pressure (which makes things go faster), they often also allow for more optimized code for all kinds of operations.

Not a change request for this PR, just a point for a future optimization.